code/game/objects/items/devices/multitool.dm

```diff
@@ -51,10 +51,10 @@
 	return ..()
 
 /obj/item/device/multitool/proc/mode_switch(mob/living/user)
-	if(++mode_index > modes.len) mode_index = 1
+	if(mode_index + 1 > modes.len) mode_index = 1

```

Comment:
The bug was originally caused by the double-increment because of the pre-increment. Basically when it was checking the old `if()` statement, it would increment `mode_index`, and then if that condition failed, it would increment it again.
i.e., just dropping the `else` clause and leaving the `if()`'s condition intact should also be a valid fix.